### PR TITLE
Continue to update unit testing and debugging MD to use VS 2017

### PR DESCRIPTION
**Pull request by eriawan**
This PR is a part of WIP of task list in #3474 to update some ASP.NET Classic related pages to use Visual Studio 2017, as agreed in proposal #3452 (closed).

This PR contains update of multiple pages but related (under the section of ASP.NET Web API 2, subsection of Testing and Debugging:

- https://github.com/aspnet/Docs/blob/master/aspnet/web-api/overview/testing-and-debugging/unit-testing-with-aspnet-web-api.md
- https://github.com/aspnet/Docs/blob/master/aspnet/web-api/overview/testing-and-debugging/unit-testing-controllers-in-web-api.md
- https://docs.microsoft.com/en-us/aspnet/web-api/overview/testing-and-debugging/mocking-entity-framework-when-unit-testing-aspnet-web-api-2

The updates were done in those pages in this PR, because all three pages are closely related.
The images have had been updated accordingly, especially on the images that show nuget installation and Test Explorer.

The sample code has been updated as well, and it's been uploaded to MSDN code gallery in this link: https://code.msdn.microsoft.com/Unit-Testing-with-ASPNET-1374bc11

Please review, thank you in advance.

### CODE REVIEW

**Comment by eriawan** [comment]
@Rick-Anderson 

Could you review this PR?

Thanks in advance.

**Comment by scottaddie** [comment]
@Rick-Anderson @eriawan I'll take care of this, since Rick is on vacation.

**Comment by eriawan** [comment]
@scottaddie Many thanks!

File: aspnet/web-api/overview/testing-and-debugging/unit-testing-controllers-in-web-api.md [hunk 1]
```diff
@@ -63,15 +65,12 @@ The `Post` method calls **UrlHelper.Link** to create links in the response. This
 
 The **UrlHelper** class needs the request URL and route data, so the test has to set values for these. Another option is mock or stub **UrlHelper**. With this approach, you replace the default value of [ApiController.Url](https://msdn.microsoft.com/en-us/library/system.web.http.apicontroller.url.aspx) with a mock or stub version that returns a fixed value.
 
-Let's rewrite the test using the [Moq](https://github.com/Moq) framework.
+Let's rewrite the test using the [Moq](https://github.com/Moq) framework. You can get Moq as nuget package by adding Moq into the test project.
```
Comment (scottaddie) [inline aspnet/web-api/overview/testing-and-debugging/unit-testing-controllers-in-web-api.md]: ~~You can get Moq as nuget package by adding Moq into the test project.~~
Install the `Moq` NuGet package in the test project.

Comment (eriawan) [reply aspnet/web-api/overview/testing-and-debugging/unit-testing-controllers-in-web-api.md]: Done updated. Thanks for the feedback!

**Comment by scottaddie** [comment]
@eriawan Thank you for your help with these updates! It's greatly appreciated by the team.

**Pull request merged**